sources/packages/backend/apps/api/src/route-controllers/application-offering-change-request/_tests_/e2e/application-offering-change-request.institutions.controller.getEligibleApplications.e2e-spec.ts

```diff
@@ -0,0 +1,203 @@
+} from "@sims/sims-db";
+import { FieldSortOrder } from "@sims/utilities";
+import { getUserFullName } from "../../../../utilities";
+
+describe("ApplicationOfferingChangeRequestInstitutionsController(e2e)-getEligibleApplications", () => {
+  let app: INestApplication;
+  let db: E2EDataSources;
+  let collegeFLocation: InstitutionLocation;
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource } = await createTestingAppModule();
+    app = nestApplication;
+    db = createE2EDataSources(dataSource);
+    // College F.
+    const { institution: collegeF } = await getAuthRelatedEntities(
+      db.dataSource,
+      InstitutionTokenTypes.CollegeFUser,
+    );
+    collegeFLocation = createFakeInstitutionLocation(collegeF);
+    await authorizeUserTokenForLocation(
+      db.dataSource,
+      InstitutionTokenTypes.CollegeFUser,
+      collegeFLocation,
+    );
+  });
+
+  it(
+    `Should return all available eligible applications that can be requested for an application ` +
+      `offering change for the location in ${FieldSortOrder.DESC} order of application number when requested.`,
+    async () => {
+      // Arrange
+
+      // Student 1 has a completed application to the institution.
+      const completedApplication = await saveFakeApplicationDisbursements(
+        db.dataSource,
+        { institutionLocation: collegeFLocation },
+        {
+          applicationStatus: ApplicationStatus.Completed,
+        },
+      );
+      // Student 2 has a completed application to the institution, which will be archived.
+      const archivedCompletedApplication =
+        await saveFakeApplicationDisbursements(
+          db.dataSource,
+          { institutionLocation: collegeFLocation },
+          {
+            applicationStatus: ApplicationStatus.Completed,
+          },
+        );
+      // Student 3 has an application waiting for assessment to the institution.
+      await saveFakeApplicationDisbursements(db.dataSource, {
+        institutionLocation: collegeFLocation,
+      });
+      // Student 4 has a completed application for a different institution.
+      await saveFakeApplicationDisbursements(db.dataSource, undefined, {
+        applicationStatus: ApplicationStatus.Completed,
+      });
+      // Student 5 has a completed application for the institution, that have a pending application offering change request.
+      await saveFakeApplicationOfferingRequestChange(db, {
+        institutionLocation: collegeFLocation,
+      });
+      // Student 6 has a completed application for the institution, that have an approved application offering change request.
+      const approvedApplicationOfferingChange =
+        await saveFakeApplicationOfferingRequestChange(
+          db,
+          {
+            institutionLocation: collegeFLocation,
+          },
+          {
+            initialValues: {
+              applicationOfferingChangeRequestStatus:
+                ApplicationOfferingChangeRequestStatus.Approved,
```

Comment:
We can also include application offering change in status declinedBySABC and declinedByStudent as they are also eligible.

Reply:
M,y intention was to show at least one `completed` status

Reply:
I think it's good to add them. In case the logic for the statuses change, the tests will complain that something is not as it was supposed to be.

Reply:
it's about adding remaining completed statuses (declined by student and SABC) and expect  them to be returned

Reply:
If there is no other test testing I would say that we can add declined by student and SABC.

Reply:
I think we can add them.